src/test/java/factories/CheckoutFactory.java

```diff
@@ -23,16 +23,38 @@ public Checkout build() {
         checkout.payment.discountCodes = new ArrayList<>();
 
         checkout.customer = new Customer();
-        //this is set with fake information for now as I just wanted to see if I had achieved the correct format for the JSON
-        checkout.customer.emailAddress = "blah@yahoo.es";
-        checkout.customer.giftOoption = GiftOption.yes;
-        checkout.customer.name = "blah";
+        checkout.customer.emailAddress = new EmailAddressFactory().build();
+        checkout.customer.giftOption = GiftOption.yes;
+        String customerName = new RandomStringFactory().build();
+        checkout.customer.name = customerName;
         checkout.customer.newsletterSignup = true;
-        checkout.customer.phoneNumber = "02086400366";
+        checkout.customer.phoneNumber = new PhoneNumberFactory().build();
         Guest guest = new Guest();
-        guest.name = "yoquese";
+        guest.name = customerName;
         checkout.customer.guests = new ArrayList<>();
         checkout.customer.guests.add(guest);
+
+        checkout.delivery = new Delivery();
+        checkout.delivery.address  = null;
+
+        checkout.offers = new ArrayList<>();
+        Offer offer = new Offer();
+        offer.fulfillment = Fulfillment.APPOINTMENT;
+        offer.date = currentDate.toString();
+        offer.time = "825";
+        offer.offerId = 1178242;
+        offer.offerType = OfferType.SERVICE;
+        offer.quantity = 1;
+        Sku sku = new Sku();
+        sku.skuId = 1317696;
+        offer.skus = new ArrayList<>();
```

Comment:
You might want to consider defining `skus` (and other collection fields) like this in the model classes:

```java
public List<Sku> skus = new ArrayList<>();
```

Then you wouldn't need to assign a new list when you want to use them which would make the test code a little neater. However, it would probably mean that the API request is serialized as `"skus": []` rather than `"skus": null` (or no `skus` field at all, depending on how the Java libraries work) which might matter, or it might not.